packages/forms/src/UIForm/fields/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -4,19 +4,12 @@ import memoize from 'lodash/memoize';
 import InputDateTimePickerComponent from '@talend/react-components/lib/DateTimePickers';
 import FieldTemplate from '../FieldTemplate';
 import { isoDateTimeRegExp } from '../../customFormats';
-import { UnhandleTypeError, UnexpectedTypeError } from './WrongTypeError';
+import { WidgetUnhandleTypeError, WidgetUnexpectedTypeError } from './WrongTypeError';
 
-export const GENERIC_FORMAT_ERROR = 'GENERIC FORMAT ERROR';
-
-const INVALID_DATE = new Date('');
 const HANDLE_CONVERTION_TYPE = ['string', 'number'];
 
-function isDateValid(date) {
-	if (date === undefined) {
-		return true;
-	}
-
-	return date instanceof Date && !isNaN(date.getTime());
+function generateInvalidDate() {
```

Comment:
Why did you replaced the const INVALID_DATE by a function that create a new one each time ? You even have a test on that, guess you really want that for a reason.

Reply:
Effectively :)
There is multiple possibilities to spread an InvalidDate to the component. When we switch for one error possibility to another still giving an InvalidDate, we couldn't figure out a change if it's the same instance.
And for the component point of view, the prop don't change from the last one, so it can't update itself.
One case :
- spread an InvalidDate due to 'error possibility X'
- define a valid date in the picker/component
- new error in the widget so spread a new InvalidDate due to 'error possibility Y'
(I lost the specific real case :/)

It's only for specific cases when errors could be raised. But I think just by principle, it's a good idea to have a different instance for different errors/data. And if it's the same data flowing into the widget, this will be memoized, so the same error and so, same instance.